python/lsst/ts/m2gui/display/figure_constant.py

```diff
@@ -123,45 +147,250 @@ def _set_default_series(self, range_x_min, range_x_max, num_points):
             Minimum of the range of x-axis.
         range_x_max : `float` or `int`
             Maximum of the range of x-axis.
-        num_points : `int`
-            Number of the points.
+        num_lines : `int`
+            Number of the lines.
+        text_legends : `list [str]`
+            Text of the legends. The number of elements should be the same as
+            num_lines. Put [] if you do not want to show the legend.
         """
 
-        # Default data to allocate the needed memory or space
-        series = QtCharts.QLineSeries()
-        for x in np.linspace(range_x_min, range_x_max, num=num_points):
-            series.append(x, 0)
-
         chart = self.chart()
-        chart.addSeries(series)
+        for idx in range(num_lines):
+
+            series = QtCharts.QLineSeries()
+
+            if not self._is_realtime:
+                # Default data to allocate the needed memory or space
+                for x in np.linspace(range_x_min, range_x_max, num=self._num_points):
+                    series.append(x, 0)
+
+            chart.addSeries(series)
+
+            series.attachAxis(self.axis_x)
+            series.attachAxis(self.axis_y)
+
+        # Update the legend
+        legend = chart.legend()
+        for marker, text_legend in zip(legend.markers(), text_legends):
```

Comment:
This is probably more complicated than needs to be. Why you don't set series name as legend? Setting series.setName above fill make legend, will be simpler, and will name serie (now series are name-less).

Reply:
Fixed in the commit: [a142e4c](https://github.com/lsst-ts/ts_m2gui/pull/14/commits/a142e4c44edda4f9ceebea4ed4a28648c7289a94). Thanks!